lib/src/main/kotlin/at/bitfire/synctools/mapping/calendar/processor/RecurrenceFieldsProcessor.kt

```diff
@@ -60,8 +70,18 @@ class RecurrenceFieldsProcessor: AndroidEventFieldProcessor {
         // EXRULE
         values.getAsString(Events.EXRULE)?.let { rulesStr ->
             try {
-                for (rule in rulesStr.split(AndroidTimeUtils.RECURRENCE_RULE_SEPARATOR))
-                    to.exRules += ExRule(null, rule)
+                for (rule in rulesStr.split(AndroidTimeUtils.RECURRENCE_RULE_SEPARATOR)) {
+                    val rule = ExRule(null, rule)
+
+                    // skip if UNTIL is before event's DTSTART
+                    val tsUntil = rule.recur.until?.time
+                    if (tsUntil != null && tsUntil <= tsStart) {
+                        logger.warning("Ignoring $rule because UNTIL ($tsUntil) is not after DTSTART ($tsStart)")
+                        continue
+                    }
+
+                    to.exRules += rule
```

Comment:
Does not seem like this code change in the for-loop is being covered by the tests.

I can replace it with the old code:
```
to.exRules += ExRule(null, rule)
```

and the tests still run green.